Connection/Connection.php

```diff
@@ -681,21 +681,31 @@ public function read($length, $flags = 0)
             );
         }
 
-        if (true === $this->isEncrypted()) {
-            return fread($this->getStream(), $length);
-        }
+        $out = '';
 
-        if (false === $this->isRemoteAddressConsidered()) {
-            return stream_socket_recvfrom($this->getStream(), $length, $flags);
-        }
+        do {
+            $buffer = null;
+            $readLength = $length - strlen($out);
 
-        $out = stream_socket_recvfrom(
-            $this->getStream(),
-            $length,
-            $flags,
-            $address
-        );
-        $this->_remoteAddress = !empty($address) ? $address : null;
+            if (0 === $readLength) {
+                break;
+            }
+
+            if (true === $this->isEncrypted()) {
+                $buffer = fread($this->getStream(), $readLength);
+
+                if ('' === $buffer) {
+                    $buffer = false;
```

Comment:
Can we use another variable to stop the loop please? Something like `$break`. Alternatively, an infinite loop with a `break` would fit too.